main.tf

```diff
@@ -263,6 +265,7 @@ resource "aws_dynamodb_table" "terraform_state_lock" {
 
   server_side_encryption {
     enabled = true
+    kms_key_arn = data.aws_iam_policy_document.policy.arn
```

Comment:
You are pointing to the incorrect resource here. You should also be able to specify the default KMS key for dynamo. It should have an alias like `aws/dynamo`

Reply:
done